docs/examples/use_cases/pytorch/efficientnet/image_classification/dali.py

```diff
@@ -19,14 +19,20 @@
 
 from nvidia.dali.auto_aug import auto_augment, trivial_augment
 
-
-@pipeline_def(enable_conditionals=True)
+@pipeline_def(enable_conditionals=True, exec_dynamic=True)
 def training_pipe(data_dir, interpolation, image_size, output_layout, automatic_augmentation,
-                  dali_device="gpu", rank=0, world_size=1):
+                  dali_device="gpu", rank=0, world_size=1, send_filepaths=False):
     rng = fn.random.coin_flip(probability=0.5)
 
-    jpegs, labels = fn.readers.file(name="Reader", file_root=data_dir, shard_id=rank,
-                                    num_shards=world_size, random_shuffle=True, pad_last_batch=True)
+    if data_dir is None:
+        if send_filepaths:
+            filepaths = fn.external_source(name="images", no_copy=True)
+            jpegs = fn.io.file.read(filepaths)
+        else:
+            jpegs = fn.external_source(name="images", no_copy=True)
+    else:
+        jpegs, labels = fn.readers.file(name="Reader", file_root=data_dir, shard_id=rank,
+                                        num_shards=world_size, random_shuffle=True, pad_last_batch=True)
```

Comment:
I don't want to say no to this, but I'm not that fond of making the quite relevant part of the example a tree of if variants. We link to this code from the example: https://docs.nvidia.com/deeplearning/dali/main-user-guide/docs/examples/use_cases/pytorch/efficientnet/readme.html#efficientnet-for-pytorch-with-dali-and-autoaugment and Imo it should not overshadow the main application of automatic augmentations.
The more we add the less it's an example and more of a test code.

Although I agree that it would be nice to show it as an option to the user and allow them to do their own benchmarking (we got messages from someone using this code for benchmarks). 

Should we maybe consider a separate file for the pipeline and the factory used with proxy?